packages/agreement-process/test/submitAgreement.test.ts

```diff
@@ -0,0 +1,1742 @@
+    await addOneAttribute(attribute);
+    await addOneAgreement(agreement);
+
+    const authData = getRandomAuthData(consumer.id);
+    const mockUserResponse: UserResponse = {
+      email: "selfcare.test.submitagreement@test.org",
+      name: "Test Name",
+      surname: "Test Surname",
+      id: generateId(),
+      taxCode: "TAXCODE",
+    };
+
+    mockSelfCareResponse(mockUserResponse);
+    const submittedAgreement = await agreementService.submitAgreement(
+      agreement.id,
+      {
+        consumerNotes: consumerNotesText,
+      },
+      {
+        authData,
+        correlationId: randomUUID(),
+        serviceName: "AgreementServiceTest",
+        logger: genericLogger,
+      }
+    );
+
+    expect(submittedAgreement).toBeDefined();
+    expect(submittedAgreement.state).toBe(agreementState.active);
+
+    const uploadedFiles = await fileManager.listFiles(
+      config.s3Bucket,
+      genericLogger
+    );
+
+    expect(submittedAgreement.contract).toBeDefined();
+    expect(uploadedFiles[0]).toEqual(submittedAgreement.contract?.path);
+
+    const actualAgreementData = await readLastAgreementEvent(agreement.id);
+    if (!actualAgreementData) {
+      fail("Creation fails: agreement not found in event-store");
+    }
+
+    expect(actualAgreementData.type).toEqual("AgreementActivated");
+    expect(actualAgreementData).toMatchObject({
+      type: "AgreementActivated",
+      event_version: 2,
+      version: "1",
+      stream_id: submittedAgreement.id,
+    });
+
+    const actualAgreement: AgreementV2 | undefined = decodeProtobufPayload({
+      messageType: AgreementActivatedV2,
+      payload: actualAgreementData.data,
+    }).agreement;
+
+    if (!actualAgreement) {
+      fail("impossible to decode AgreementAddedV1 data");
+    }
+
+    expect(selfcareV2ClientMock.getUserInfoUsingGET).toHaveBeenCalled();
+    expect(actualAgreement).toMatchObject({
+      id: submittedAgreement.id,
+      eserviceId: eservice.id,
+      descriptorId: descriptor.id,
+      producerId: producer.id,
+      consumerId: consumer.id,
+      state: toAgreementStateV2(agreementState.active),
+      contract: {
+        id: expect.any(String),
+        name: expect.any(String),
+        prettyName: expect.any(String),
+        contentType: expect.any(String),
+        path: expect.any(String),
+        createdAt: expect.any(BigInt),
+      },
+      consumerNotes: consumerNotesText,
+      verifiedAttributes: [
+        {
+          id: validVerifiedTenantAttribute.id,
+        },
+      ],
+      suspendedByConsumer: false,
+      suspendedByProducer: false,
+      suspendedByPlatform: false,
+    });
+    expect(actualAgreement.stamps?.activation).toMatchObject({
+      who: authData.userId,
+      when: expect.any(BigInt),
+    });
+  });
+
+  it("should submit agreement contract with new state PENDING when producer is different from consumer and not related agreements, generates AgreementSubmitted event", async () => {
```

Comment:
is this what you meant?
```suggestion
  it("should submit agreement with new state PENDING when producer is different from consumer and no related agreements exist, generates AgreementSubmitted event", async () => {
```